### PR TITLE
feat(16): added email reset [server]

### PR DESCRIPTION
**Pull request by dartwlad**
suggested flow
- user invoke GET /api/management/email-reset with Authorization header (token from login response is a value) and gets an email with redirection link
- user enters page and sending new email with token POST /api/management/email-update {emailResetToken: string, public email: string}
- In response, the user gets updated profile with a new token

### CODE REVIEW

File: packages/server/migrations/1594758731889-migration.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
```
Comment (karolinayanko) [inline packages/server/migrations/1594758731889-migration.ts]: spaces between {} is missing

**Review by karolinayanko — APPROVED** [review]

File: packages/server/migrations/1594758731889-migration.ts (hunk 1 above)
Comment (dubbha) [reply packages/server/migrations/1594758731889-migration.ts]: Is lint really running for server? I don't think it should  pass

File: packages/server/package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "",
   "main": "src/server.ts",
   "scripts": {
-    "lint": "eslint . --ext .ts",
+    "lint": "eslint ./src --ext .ts",
```
Comment (dubbha) [inline packages/server/package.json]: Aha, so you decided not to lint the migrations

File: packages/server/migrations/1594758731889-migration.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+export class migration1594758731889 implements MigrationInterface {
```
Comment (dubbha) [inline packages/server/migrations/1594758731889-migration.ts:3]: unix timestamp?
don't we want to version the migrations?

File: packages/server/src/authentication/authentication.service.ts [hunk 4]
```diff
@@ -23,6 +22,8 @@ import PasswordUpdateDto from './password-update.dto';
 import WrongPasswordResetToken from '../exceptions/wrong-password-reset-token.exception';
 import PasswordResetTokenExpired from '../exceptions/password-reset-token-expired.exception';
 import { createNewPassword } from '../utils/create-new-password';
+import { createExpiresInHours } from '../utils/create-expires-in-hourse';
```
Comment (dubbha) [inline packages/server/src/authentication/authentication.service.ts:25]: Typo, should be `create-expires-in-horse`

**Review by dubbha — APPROVED** [review]

**Pull request merged**